src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -121,10 +191,32 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
    */
   protected function getUserGroups() {
     $user = User::load($this->currentUser->id());
-    /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
-    $membership_manager = \Drupal::service('og.membership_manager');
-    $other_groups = $membership_manager->getUserGroups($user);
+    $other_groups = $this->OgMembershipManager->getUserGroups($user);
     return isset($other_groups[$this->configuration['target_type']]) ? $other_groups[$this->configuration['target_type']] : [];
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    // Filter out the bundles that are not groups.
+    $entity_type_id = $this->configuration['target_type'];
+    $entity_type = $this->entityManager->getDefinition($entity_type_id);
+    $bundles_info = $this->entityManager->getBundleInfo($entity_type_id);
+
+    if ($entity_type->hasKey('bundle')) {
+
+      foreach (Og::groupTypeManager()->getGroupBundlesByEntityType($entity_type_id) as $bundle) {
```

Comment:
It's not a costly operation but still, I'd move getGroupBundlesByEntityType() out of the foreach loop for readability:
```suggestion
      $bundles = Og::groupTypeManager()->getGroupBundlesByEntityType($entity_type_id);
      foreach ($bundles as $bundle) {
```